packages/client/src/components/RoomTable.tsx

```diff
@@ -20,20 +22,27 @@ export const RoomTable = () => {
     return () => {
       gameIO.off(RoomEvent.Update, handleRoomUpdate);
     };
-  }, []);
+  }, [history]);
+
+  const joinRoom = (roomName: string) => {
+    history.push(`/game/${roomName}`);
+  };
 
   return (
     <Box>
       <Heading as="h2">Rooms</Heading>
       {rooms.map((room) => (
         <Card key={room.name} mt={1}>
-          <Flex sx={{ flexDirection: 'column' }}>
-            <Box>
-              Room: <strong>{room.name}</strong>
-            </Box>
-            <Box>
-              Players: <strong>{room.players.map((player) => player.name).join(', ')}</strong>
-            </Box>
+          <Flex sx={{ justifyContent: 'space-between' }}>
+            <Flex sx={{ flexDirection: 'column' }}>
+              <Box>
+                Room: <strong>{room.name}</strong>
+              </Box>
+              <Box>
+                Players: <strong>{room.players.map((player) => player.name).join(', ')}</strong>
+              </Box>
+            </Flex>
+            <Button onClick={() => joinRoom(room.name)}>Join</Button>
```

Comment:
why not just making it a link?

Reply:
Don't know, thought history is better for SPA...? Done.